renormalizer/property/property.py

```diff
@@ -0,0 +1,66 @@
+from typing import Union, List, Dict
+from renormalizer.mps import Mpo, Mps, MpDm
+
+class Property():
+    '''
+    class to store the properties mpos and expectation results.
+    
+    Property can be added in any TdMpsJob class (see example in thermalprop) to define
+    the user-defined mpos whose expectation values should be calculated and
+    stored during TD simulation.
+    Note that when the result is complex number, use dump_type=".npz" in TdMpsJob.
+    '''
+
+    def __init__(self, prop_strs:List[str], prop_mpos:Dict[str, Mpo]):
+        '''
+        prop_strs: names of the properties
+        prop_mpos: dict: mpos of the properties
+        '''
+        self.prop_strs = prop_strs
+        self.prop_mpos = prop_mpos
+        
+        self.prop_res = {}
+        for prop_str in prop_strs:
+            self.prop_res[prop_str] = []
+        
+
+    def calc_properties(self, mpss: Union[Dict[str,Mps], Dict[str,MpDm],  Mps, MpDm], 
+            mps_conjs:Union[Dict[str, Mps], Dict[str, MpDm], Mps, MpDm, None] =None):
+        '''
+        calculate all the properties with same mps or mps_conj
+        or 
+        calculate each property with different {prop_str:mps}, {prop_str:mps_conj}
+        '''
+        for prop_str in self.prop_strs:
+            
+            # pick ket
+            if mpss.is_mps or mpss.is_mpdm:
```

Comment:
If `Dict[str, Mps]` is passed in as `mpss` then this line would result in an `AttributeError`.  I suggest deleting the logic for `Dict` and create a new method when the necessity comes.

Reply:
Yes. It's a bug.

Reply:
fixed